code-challenges.test.js

```diff
@@ -22,25 +35,101 @@ const hitchhikersCharacters = [
+// Passing the object along with the key and value through the map method allows us to have access to each key value pair.
+// The final outcome is a sentence so we can use string interpolation to us the key value pairs in a string. 
+// Since the name needs to be capitalized, we can split the name into an array, map through the array to access the words first character (charAt) located at the 0 index and turn that into an uppercase. The return of that would not include the rest of the word so we add .slice(1) to add the rest of the word back onto uppercased character. 
+// We need to get back into a string, so we can use the .join method to get there. 
+// Complete the sentence using the interpolation.
+// Return the overall output. 
+
+
+const arrSentence = (arr) => {
+  return arr.map(({name, occupation}) => `${name.split(" ").map(word => word.charAt(0).toUpperCase() + word.slice(1)).join(" ")} is ${occupation}.`
+)}
+
+// PASS  ./code-challenges.test.js
+// arrSentence
+//   ✓ Takes in an array of objects and returns an array with a sentence about each person with their name capitalized. (3 ms)
+
+
 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
 // a) Create a test with an expect statement using the variables provided.
+describe("divRemainder", () => {
+  it("takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.", () => {
+    expect(divRemainder(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+    expect(divRemainder(hodgepodge2)).toEqual([ 2, 1, -1 ])
+  })
+})
+// ● divRemainder › takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
+
+// ReferenceError: divRemainder is not defined
 
 const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
 // Expected output: [ 2, 0, -1, 0 ]
 const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
 // Expected output: [ 2, 1, -1 ]
 
 // b) Create the function that makes the test pass.
+// Pseudocode
+// Input: An array with mixed data types
+// Output: An array of only the REMAINDERS of the numbers when divided by 3
+// Process:
+// Create a function called divRemainder
+// We know the outcome is going to be a subset of the original array of different length so we can use the filter method to iterate through the original array.
+// We only want nummbers so we can set a condition that we are keeping only the values that have the typeof 'number' or the value of 0 itself. 
+// Once we have the original numbers we can map over the new array and return each value modulus 3 to get the remainder of each value and update the array. 
+// Returns expected output.
+
+const divRemainder = (arr) => {
+    return arr.filter((value) => typeof value === 'number' || value === 0).map((value) => {return value % 3})
```

Comment:
Great logic. 0 is considered a number, so no need for the logical OR operator with the second condition there.